src/electron/calc_checksum.action.mjs

```diff
@@ -0,0 +1,65 @@
+// Copyright 2022 The Outline Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// This action will calculate SHA256 checksums based on the target platform:
+//   - Linux: calculate checksums for all files in 'tools/outline_proxy_controller/dist/'
+//   - Windows: do nothing
+// The checksums will be put in 'build/checksums/$filename.$fileext.sha256'.
+// Then the 'build/checksums' folder should be included in 'electron-builder.json'.
+
+import crypto from 'node:crypto';
+import {mkdir, readFile, writeFile} from 'node:fs/promises';
+import {basename, join as pathJoin, resolve} from 'node:path';
+import url from 'node:url';
+
+import {getRootDir} from '../build/get_root_dir.mjs';
+import {getElectronBuildParameters} from './get_electron_build_parameters.mjs';
+
+const TARGET_FOLDER = pathJoin('build', 'checksums');
+
+const LINUX_FILES_TO_HASH = [
+  pathJoin('tools', 'outline_proxy_controller', 'dist', 'install_linux_service.sh'),
+  pathJoin('tools', 'outline_proxy_controller', 'dist', 'outline_proxy_controller.service'),
+  pathJoin('tools', 'outline_proxy_controller', 'dist', 'OutlineProxyController'),
+];
+
+async function calculate_sha256_checksum(fileToHashInRelativePath) {
+  const root = getRootDir();
+  const srcFile = resolve(root, fileToHashInRelativePath);
+  const targetFile = resolve(root, pathJoin(TARGET_FOLDER, `${basename(srcFile)}.sha256`));
+  await mkdir(resolve(root, TARGET_FOLDER), {recursive: true});
+
+  const content = await readFile(srcFile);
+  const checksum = crypto
+    .createHash('sha256')
+    .update(content)
+    .digest('hex');
+  await writeFile(targetFile, checksum);
+  console.info(`calculated sha256 for "${srcFile}"`);
+}
+
+export async function main(...parameters) {
```

Comment:
Why call this as a separate binary when you can call it as a library from the node.js code?

You could run the function on the FUSE file to get the baseline hash, then again on the imutable copy of the file, and compare them.

Reply:
👍 Sounds good, so we don't need any additional build steps.